model/common_structures.smithy

```diff
@@ -59,3 +59,29 @@ structure Hits{
     _source: Document,
     fields: Document
 }
+
+structure Tenant{
+    reserved: Boolean,
+    hidden: Boolean,
+    description: String,
+    static: Boolean
+}
+
+list TenantList{
+    member: Tenant
+}
+
+map AttributeMap {
+    key: String,
+    value: String
+}
+
+list PatchOperationList{
+    member: PatchOperation
+}
+
+structure PatchOperation {
+    op: String,
+    path: String,
+    value: AttributeMap
+}
```

Comment:
This file will grown extremely large if we put every structure here. It's only meant for structure to be reused like the results of different types of search. Can we move these into the structures.smithy file of the respective operation?